stdlib/public/core/LazyCollection.swift

```diff
@@ -10,8 +10,32 @@
 //
 //===----------------------------------------------------------------------===//
 
-@available(swift, deprecated: 5.0, renamed: "LazySequence")
-public typealias LazyCollectionProtocol = LazySequenceProtocol
+public protocol _LazyCollectionProtocol: Collection, LazySequenceProtocol {		
+   /// A `Collection` that can contain the same elements as this one,		
+   /// possibly with a simpler type.		
+   ///		
+   /// - See also: `elements`		
+   associatedtype Elements : Collection = Self		
```

Comment:
Shouldn't this be a restriction on the existing associated type, rather than declaring a new one?

Reply:
Yeah, you're right, works fine. I guess that's at least something even if the protocol comes back.